test/integration/client-side-encryption/client_side_encryption.prose.26.custom_aws_credential_providers.test.ts

```diff
@@ -0,0 +1,109 @@
+import { expect } from 'chai';
+
+/* eslint-disable @typescript-eslint/no-restricted-imports */
+import { ClientEncryption } from '../../../src/client-side-encryption/client_encryption';
+import { AWSTemporaryCredentialProvider, Binary, MongoClient } from '../../mongodb';
+import { getEncryptExtraOptions } from '../../tools/utils';
+
+const metadata: MongoDBMetadataUI = {
+  requires: {
+    clientSideEncryption: true,
+    mongodb: '>=4.2.0',
+    topology: '!load-balanced'
+  }
+} as const;
+
+const masterKey = {
+  region: 'us-east-1',
+  key: 'arn:aws:kms:us-east-1:579766882180:key/89fcc2c4-08b0-4bd9-9f25-e30687b580d0'
+};
+
+describe('25. Custom AWS Credential Providers', metadata, () => {
+  let keyVaultClient;
+  let credentialProvider;
+
+  beforeEach(async function () {
+    this.currentTest.skipReason = !AWSTemporaryCredentialProvider.isAWSSDKInstalled
+      ? 'This test must run in an environment where the AWS SDK is installed.'
+      : undefined;
+    this.currentTest?.skipReason && this.skip();
+
+    keyVaultClient = this.configuration.newClient(process.env.MONGODB_UR);
+    // @ts-expect-error We intentionally access a protected variable.
+    credentialProvider = AWSTemporaryCredentialProvider.awsSDK;
+  });
+
+  afterEach(async () => {
+    await keyVaultClient?.close();
+  });
+
+  context(
+    'Case 1: Explicit encryption with credentials and custom credential provider',
+    metadata,
+    function () {
+      it('throws an error', metadata, function () {
+        expect(() => {
+          new ClientEncryption(keyVaultClient, {
+            keyVaultNamespace: 'keyvault.datakeys',
+            kmsProviders: {
+              aws: {
+                accessKeyId: process.env.FLE_AWS_KEY,
+                secretAccessKey: process.env.FLE_AWS_SECRET
+              }
+            },
+            credentialProviders: { aws: credentialProvider.fromNodeProviderChain() }
+          });
+        }).to.throw(/custom credential provider and credentials/);
+      });
+    }
+  );
+
+  context('Case 2: Explicit encryption with custom credential provider', metadata, function () {
+    let clientEncryption;
+
+    beforeEach(function () {
+      const options = {
+        keyVaultNamespace: 'keyvault.datakeys',
+        kmsProviders: { aws: {} },
+        credentialProviders: {
+          aws: async () => {
```

Comment:
I think we might need sinon here too.  Loading credentials from secrets manager locally, I see `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` in the environment, set to the same values as `FLE_AWS_KEY` and `FLE_AWS_SECRET`.  This is being used currently for automatic AWS KMS credential refresh in our legacy and unified tests, I believe. 

(we should also update the spec with this change)

Reply:
I've updated to check call count here as well.